### ISSUE TITLE
Difficulties with updating and testing website pages

### ISSUE THREAD

**Issue by duncombe** [ioos/sos-guidelines-hugo#6]
I have finally dealt with @abirger's pull request for sos-guidelines. Once I had dealt with merging the new content into a branch representing master, the difficulty was getting the updated gh-pages branch to overwrite the old one. Either I don't understand how subtree works (quite likely!) or there is something wrong with the algorithm. Blindly following the formula outlined in the hugo pages (and as interpreted by Alex in the documentation) created apparently insurmountable complaints about the pushed tip being behind however which way I twisted it. Eventually I manually bent the recalcitrant branch to my will, without the use of subtree, and the website (gh-pages branch) is now live consistent with the source material in master. Interested to see how another update proceeds, but the documentation for building the website, or updating the algorithm will need some attention eventually. [@dpsnowden, @duncombe]

**Comment by duncombe** [comment]
One problem I see with the algorithm (both in Hugo documentation and sos-guidelines Website_Deployment_Workflow.md) is that unless all files are actually removed in gh-pages branch, instead of only from the index, then the subsequent git checkout master complains. So at least one step,  or command option is missing.

**Comment by duncombe** [comment]
Another problem: once the public/ directory is removed (`rm -rf public`)  the next command in the algorithm (`git subtree add`) fails because working tree has modifications. Another command missing from the algorithm. The result of the `rm` must be committed.

**Comment by abirger** [comment]
@duncombe , really strange about subtree, it works like a charm for me...

In regard to `rm -rf public`, the reason for the fail is that it actually is just `rm -rf`, not `git rm`. As I have Windows, I just deleted the public folder altogether on my local drive, and then the `git subtree` gave me no problem at all...

**Comment by duncombe** [comment]
Yes, I spotted the mistake in my comment and fixed it. And yes, I had/have problems removing the files and directory without telling git about it.

**Comment by duncombe** [comment]
Another problem, step 9, the command as given causes a complaint 
`You must provide <repository> <ref>`
and will not proceed without adding the full repository and branch information.  Edit command to reflect that.

**Comment by abirger** [comment]
Step 9 is kind of optional, isn't it? So, I've never actually tried to fix it, and marked is as optional in the Workflow edition that I included into the "conventions-for-observing-asset-identifiers' repo pull request. Perhaps, we should find a dedicated place for the Workflow to keep working on it.

**Comment by duncombe** [comment]
Nothing wrong with keep working on it where it is for now. However, it is
now in at least two places, so yes, it should probably go into some common
area and be cloned into any repo that refers to it to keep it consistent
across the suite.

On Fri, Nov 14, 2014 at 12:27 PM, abirger notifications@github.com wrote:

> Step 9 is kind of optional, isn't i? So, I've never actually tried to fix
> it, and marked is as optional in the Workflow edition that I included into
> the "conventions-for-observing-asset-identifiers' repo pull request.
> Perhaps, we should find a dedicated place for the Workflow to keep working
> on it.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ioos/sos-guidelines/issues/6#issuecomment-63099345.

## 

## 

## 

# 

Dr. Christopher M. Duncombe Rae       c deirdre.byrne@noaa.gov
hristopher.duncombe.rae@noaa.gov
Oceanographer / Data Scientist
IOOS/NOAA, Suite 1225, 1100 Wayne Avenue, Silver Spring, MD 20910, USA
Tel: +1-301-427-2450     Fax:  +1-301-427-2073
